packages/-build-infra/src/addon-build-config-for-data-package.js

```diff
@@ -1,13 +1,72 @@
 const calculateCacheKeyForTree = require('calculate-cache-key-for-tree');
+const Funnel = require('broccoli-funnel');
+const merge = require('broccoli-merge-trees');
+const BroccoliDebug = require('broccoli-debug');
+const version = require('./create-version-module');
+const { isInstrumentedBuild } = require('./cli-flags');
+const rollupPrivateModule = require('./utilities/rollup-private-module');
 
-function addonBuildConfigForDataPackage(name) {
+function isProductionEnv() {
+  let isProd = /production/.test(process.env.EMBER_ENV);
+  let isTest = process.env.EMBER_CLI_TEST_COMMAND;
+
+  return isProd && !isTest;
+}
+
+function addonBuildConfigForDataPackage(PackageName) {
   return {
-    name,
+    name: PackageName,
+
+    init() {
+      this._super.init && this._super.init.apply(this, arguments);
+      this._prodLikeWarning();
+      this.debugTree = BroccoliDebug.buildDebugCallback(PackageName);
```

Comment:
```suggestion
      this.debugTree = BroccoliDebug.buildDebugCallback(`ember-data:${PackageName}`);
```

Reply:
I did this elsewhere: https://github.com/emberjs/data/pull/6180/commits/498834f71507a4603861abe22bf82b3693e32c63#diff-2658ff1b67b663873e5835e7394a394aR14

and it should be done here too. The packages will have `@ember-data/` in their name but they wouldn't otherwise share a debug namespace and for niceness they should